chans/chans.go

```diff
@@ -176,8 +176,17 @@ func RangeBig(from, to *big.Int) <-chan *big.Int {
 // Returns number of items drained.
 func Drain[T any](channel <-chan T) int {
 	drained := 0
-	for range channel {
-		drained++
+
+	for {
+		select {
+		case _, ok := <-channel:
+			if ok {
+				drained++
+			} else {
+				return drained
+			}
+		default:
```

Comment:
I think this should be at least optional. Some callers might want to wait for messages to arrive on the channel and some might want to skip. 

That's a theoretical case, but a more real-life case can be:
- We call `Drain`
- Goroutines that write to `channel` still have finite number of items to send before they exit. 
- If they send to the channel with some delay, we can exit from this loop before everything has been drained.